cpp/src/search/search.cu

```diff
@@ -100,46 +100,43 @@ std::unique_ptr<column> search_ordered(table_view const& t,
 
   // This utility will ensure all corresponding dictionary columns have matching keys.
   // It will return any new dictionary columns created as well as updated table_views.
-  auto matched  = dictionary::detail::match_dictionaries({t, values}, stream);
-  auto d_t      = table_device_view::create(matched.second.front(), stream);
-  auto d_values = table_device_view::create(matched.second.back(), stream);
-  auto count_it = thrust::make_counting_iterator<size_type>(0);
-
-  rmm::device_vector<order> d_column_order(column_order.begin(), column_order.end());
-  rmm::device_vector<null_order> d_null_precedence(null_precedence.begin(), null_precedence.end());
+  auto const [_, t_vals_matched] = dictionary::detail::match_dictionaries({t, values}, stream);
```

Comment:
This will cause the first pair item to be destroyed on return. This is not good since the second pair item is contains `column_views` that point to the `columns` in the first pair. Using the `_` here means the `column_views` will be pointing to invalid memory/data after this line.

Assigning it to a variable will keep the column data around for the same lifetime as the views as in the original code.
```suggestion
  auto const [matched, t_vals_matched] = dictionary::detail::match_dictionaries({t, values}, stream);
```

Reply:
In C++, underscore `_` is a valid variable name, so the variable will not be destroyed untill the end of the function. That is why I have to use `_`, `__`, `___` etc.. in places of unused variables.